lib/src/switch/Switch.tsx

```diff
@@ -40,84 +39,56 @@ const DxcSwitch = ({
-          onChange(isChecked);
-        }
+        onChange?.(isChecked);
         break;
     }
   };
 
   const handlerSwitchChange = (event) => {
-    if (checked === undefined) {
-      const isChecked = event.target.checked ?? !innerChecked;
-      setInnerChecked(isChecked);
-      if (typeof onChange === "function") {
-        onChange(isChecked);
-      }
-    } else {
-      if (typeof onChange === "function") {
-        onChange(!checked);
-      }
-    }
+    checked ?? setInnerChecked((innerChecked) => !innerChecked);
+    onChange?.(checked ? !checked : !innerChecked);
   };
 
   return (
     <ThemeProvider theme={colorsTheme.switch}>
-      <SwitchContainer margin={margin} size={size} onKeyDown={handleOnKeyDown}>
-        {labelPosition === "before" && hasLabel && (
+      <SwitchContainer
+        margin={margin}
+        size={size}
+        onKeyDown={handleOnKeyDown}
+        disabled={disabled}
+        onClick={!disabled ? handlerSwitchChange : undefined}
+      >
+        {labelPosition === "before" && label && (
           <LabelContainer
             id={labelId}
             labelPosition={labelPosition}
-            onClick={!disabled ? handlerSwitchChange : undefined}
             disabled={disabled}
             backgroundType={backgroundType}
-            hasLabel={hasLabel}
+            label={label}
           >
             {label} {optional && <>{translatedLabels.formFields.optionalLabel}</>}
           </LabelContainer>
         )}
-        <SwitchBase
-          labelPosition={labelPosition}
-          hasLabel={hasLabel}
-          htmlFor={labelId}
-          onClick={disabled === true ? () => {} : handlerSwitchChange}
-        >
-          <SwitchInput
-            type="checkbox"
+        <ValueInput type="checkbox" name={name} aria-hidden={true} value={value} disabled={disabled} readOnly />
+        <SwitchBase>
+          <SwitchTrack
             role="switch"
-            name={name}
-            id={labelId}
+            backgroundType={backgroundType}
+            aria-checked={checked ?? innerChecked}
+            aria-disabled={disabled}
             disabled={disabled}
-            value={value}
             aria-labelledby={labelId}
-            aria-label={hasLabel ? label : undefined}
-            aria-checked={checked ?? innerChecked}
-            defaultChecked={defaultChecked ?? undefined}
-            tabIndex={-1}
-          ></SwitchInput>
-          {disabled ? (
-            <DisabledSwitchTrack
-              backgroundType={backgroundType}
-              data-checked={checked ?? (innerChecked ? innerChecked : undefined)}
-              tabIndex={-1}
-            />
-          ) : (
-            <SwitchTrack
-              backgroundType={backgroundType}
-              data-checked={checked ?? (innerChecked ? innerChecked : undefined)}
-              tabIndex={tabIndex}
-              ref={refTrack}
-            />
-          )}
+            tabIndex={tabIndex ? tabIndex : -1}
```

Comment:
I don't understand very well this piece of code: `tabIndex={tabIndex ? tabIndex : -1}`. Wouldn't it be `tabIndex={disabled ? -1 : tabIndex}`? Currently the component is not focusable by keyboard.

Reply:
Done